lib/std/child_process.zig

```diff
@@ -164,7 +164,10 @@ pub const ChildProcess = struct {
             return term;
         }
 
-        try windows.TerminateProcess(self.handle, exit_code);
+        windows.TerminateProcess(self.handle, exit_code) catch |err| switch (err) {
+            error.PermissionDenied => return error.AlreadyTerminated,
```

Comment:
I dont think that all PermissionDenied errors map to AlreadyTerminated cases, as the child may also simply have insufficient permissions (ie to kill a system process as user process).

Reply:
AFAICT, there are exactly two cases where ERROR_ACCESS_DENIED can be returned from TerminateProcess:

1. The process has already terminated (what we're expecting)
2. The process handle does not have the PROCESS_TERMINATE access right

In this case we know `self.handle` is the process handle that we got when creating the child process via CreateProcessW.  As noted [here](https://learn.microsoft.com/en-us/windows/win32/procthread/process-security-and-access-rights), "The handle returned by the CreateProcess function has PROCESS_ALL_ACCESS access to the process object".  Theoretically I think it's possible that a user could manually call SetSecurityInfo on the ChildProcess.handle and revoke it's PROCESS_TERMINATE, or replace the handle with an entirely different one, but at that point, I think we have to trust that the user really wants to shoot themselves in the foot.

Reply:
Fair enough. A short hint doesn't sound too bad either, because hanging processes are quite aweful: https://community.osr.com/discussion/279421/removing-process-terminate-flag-causes-hang

Reply:
Interesting, I didn't realize that the child process itself could modify the handle returned back to parent process.  Even if this is a really obscure edge case, it seems worth addressing.  I added a secondary check by calling WaitForSingleObjectEx with a 0 timeout, which should return WAIT_TIMEOUT immediately if the process is still running, or WAIT_OBJECT_0 if it really has terminated, but in either case, it shouldn't block.